nbconvert/preprocessors/execute.py

```diff
@@ -248,6 +250,16 @@ def start_new_kernel(self, **kwargs):
         kc : KernelClient
             Kernel client as created by the kernel manager `km`.
         """
+        
+        # Because of an old API, an empty kernel string should be interpreted as indicating
+        # that you want to use the notebook's metadata specified kernel.
+        # We use find_kernel_name to do this.
+        if self.kernel_name == u"":
```

Comment:
Would it be better to do `if not self.kernel_name:`? As a python developer I would expect that if "" is treated as falsy for a condition that None would be too. Or does the kernel_manager_class already handle None separately?

```
if not self.kernel_name:
  self.kernel_name = find_kernel_name(self.nb)
  if isinstance(self.kernel_name, string_types):
    warnings.warn(...)
```

Reply:
After looking at it, I'm pretty sure `None` would have worked previously… but I still don't want to encourage passing people to do this regardless. So we'll want to warn no matter what type the value was.

Reply:
So would checking for false kernel_name and declaring it deprecated in all cases make more sense from a function perspective then? I'm ok with merging if you feel strongly to just check for unicode string, it just had some code smell for something that might fail under similar situations -- like python version differences